nautobot/ipam/filters.py

```diff
@@ -510,11 +518,39 @@ class VLANFilterSet(NautobotFilterSet, LocatableModelFilterSetMixin, TenancyFilt
         to_field_name="slug",
         label="Role (slug)",
     )
+    prefixes = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=Prefix.objects.all(),
+        label="Prefix (name or ID)",
+    )
+    has_prefixes = RelatedMembershipBooleanFilter(
+        field_name="prefixes",
+        label="Has prefixes",
+    )
+    interfaces_as_untagged = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=Interface.objects.all(),
+        to_field_name="untagged_vlan_id",
+        label="Interfaces as untagged (name or ID)",
+    )
+    interfaces_as_tagged = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=Interface.objects.all(),
+        to_field_name="tagged_vlans",
+        label="Interfaces as tagged (name or ID)",
+    )
+    vminterfaces_as_untagged = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=VMInterface.objects.all(),
+        to_field_name="untagged_vlan_id",
+        label="VM Interfaces as untagged (name or ID)",
+    )
+    vminterfaces_as_tagged = NaturalKeyOrPKMultipleChoiceFilter(
+        queryset=VMInterface.objects.all(),
+        to_field_name="tagged_vlans",
+        label="VM Interfaces as tagged (name or ID)",
+    )
     tag = TagFilter()
 
     class Meta:
         model = VLAN
-        fields = ["id", "vid", "name"]
+        fields = "__all__"
```

Comment:
`__all__` while is ideal and sadly may be a best practice the serializer autodoc doesn't sit well with fields that exist but aren't really best suited to be exposed. Please update this to be explicit and add the newly created fields.